xarray/core/dataarray.py

```diff
@@ -816,6 +816,118 @@ def swap_dims(self, dims_dict):
         ds = self._to_temp_dataset().swap_dims(dims_dict)
         return self._from_temp_dataset(ds)
 
+    def set_index(self, indexers=None, append=False, inplace=False,
+                  **kw_indexers):
+        """Set DataArray (multi-)indexes using one or more existing coordinates.
+
+        Parameters
+        ----------
+        indexers : dict, optional
```

Comment:
Instead of `array.set_index({'record': ['level_0', 'level_1']})` or `array.set_index(record=['level_0', 'level_1'])`, it might suffice to simply use `array.set_index(['level_0', 'level_1'])`, without specifying which dimension these get added to. We can infer the dimension names from the variables.

The upside is that accepting a list instead of a dict is a little more succinct. The downside is that it's less explicit/self-documenting.